omp/test/matrix/dense_kernels.cpp

```diff
@@ -275,6 +281,174 @@ TEST_F(Dense, AdvancedApplyIsEquivalentToRef)
+    srmtx->convert_to(drmtx.get());
+    somtx->convert_to(domtx.get());
+
+    ASSERT_MTX_NEAR(drmtx, domtx, 1e-14);
+}
+
+
+TEST_F(Dense, MoveToCsrIsEquivalentToRef)
+{
+    auto rmtx = gko::initialize<Mtx>({{1.0, 2.0, 3.0}, {0.0, 1.5, 0.0}}, ref);
+    auto omtx = Mtx::create(omp);
+    omtx->copy_from(rmtx.get());
+
+    auto srmtx = gko::matrix::Csr<>::create(ref);
+    auto somtx = gko::matrix::Csr<>::create(omp);
+
+    rmtx->move_to(srmtx.get());
+    omtx->move_to(somtx.get());
+
+    auto drmtx = Mtx::create(ref);
+    auto domtx = Mtx::create(omp);
+    srmtx->move_to(drmtx.get());
+    somtx->move_to(domtx.get());
+
+    ASSERT_MTX_NEAR(drmtx, domtx, 1e-14);
+}
+
+
+TEST_F(Dense, ConvertToEllIsEquivalentToRef)
+{
+    auto rmtx = gko::initialize<Mtx>({{1.0, 2.0, 3.0}, {0.0, 1.5, 0.0}}, ref);
+    auto omtx = Mtx::create(omp);
+    omtx->copy_from(rmtx.get());
+
+    auto srmtx = gko::matrix::Ell<>::create(ref);
+    auto somtx = gko::matrix::Ell<>::create(omp);
+
+    rmtx->convert_to(srmtx.get());
+    omtx->convert_to(somtx.get());
+
+    auto drmtx = Mtx::create(ref);
+    auto domtx = Mtx::create(omp);
+    srmtx->convert_to(drmtx.get());
+    somtx->convert_to(domtx.get());
+
+    ASSERT_MTX_NEAR(drmtx, domtx, 1e-14);
+}
+
+
+TEST_F(Dense, MoveToEllIsEquivalentToRef)
+{
+    auto rmtx = gko::initialize<Mtx>({{1.0, 2.0, 3.0}, {0.0, 1.5, 0.0}}, ref);
+    auto omtx = Mtx::create(omp);
+    omtx->copy_from(rmtx.get());
+
+    auto srmtx = gko::matrix::Ell<>::create(ref);
+    auto somtx = gko::matrix::Ell<>::create(omp);
+
+    rmtx->move_to(srmtx.get());
+    omtx->move_to(somtx.get());
+
+    auto drmtx = Mtx::create(ref);
+    auto domtx = Mtx::create(omp);
+    srmtx->move_to(drmtx.get());
+    somtx->move_to(domtx.get());
+
+    ASSERT_MTX_NEAR(drmtx, domtx, 1e-14);
+}
+
+
+TEST_F(Dense, ConvertToHybridIsEquivalentToRef)
+{
+    auto rmtx = gko::initialize<Mtx>({{1.0, 2.0, 3.0}, {0.0, 1.5, 0.0}}, ref);
+    auto omtx = Mtx::create(omp);
+    omtx->copy_from(rmtx.get());
+
+    auto srmtx = gko::matrix::Hybrid<>::create(ref);
+    auto somtx = gko::matrix::Hybrid<>::create(omp);
+
+    rmtx->convert_to(srmtx.get());
+    omtx->convert_to(somtx.get());
+
+    auto drmtx = Mtx::create(ref);
+    auto domtx = Mtx::create(omp);
+    srmtx->convert_to(drmtx.get());
+    somtx->convert_to(domtx.get());
+
+    ASSERT_MTX_NEAR(drmtx, domtx, 1e-14);
+}
+
+
+TEST_F(Dense, MoveToHybridIsEquivalentToRef)
+{
+    auto rmtx = gko::initialize<Mtx>({{1.0, 2.0, 3.0}, {0.0, 1.5, 0.0}}, ref);
+    auto omtx = Mtx::create(omp);
+    omtx->copy_from(rmtx.get());
+
+    auto srmtx = gko::matrix::Hybrid<>::create(ref);
+    auto somtx = gko::matrix::Hybrid<>::create(omp);
+
+    rmtx->move_to(srmtx.get());
+    omtx->move_to(somtx.get());
+
+    auto drmtx = Mtx::create(ref);
+    auto domtx = Mtx::create(omp);
+    srmtx->move_to(drmtx.get());
+    somtx->move_to(domtx.get());
+
+    ASSERT_MTX_NEAR(drmtx, domtx, 1e-14);
+}
+
+
```

Comment:
I think it would be better to have tests similar to the CUDA tests, where the data is generated randomly and bigger in general. But I think we should decide on that together, what do you think, @ginkgo-project/developers ?

Reply:
That should be the case yes. So far I believe this is what we have done. This also helps in finding slow kernels (you only have to look at the time the tests take). COO is slow btw.

Reply:
I agree that large tests make sense for getting a grasp of the runtime. However, race conditions more likely show up for small matrices, correct?

Reply:
Well, anyway for debugging threading problems we need to setup something proper such as using the [ThreadSanitizer](https://clang.llvm.org/docs/ThreadSanitizer.html). I believe GCC also has one.

I believe this can be done from CTest. I need to setup a nightly run which checks for memory access and threading problems.